prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -150,21 +151,22 @@ class ReportFunction(
                 // track the sending organization and client based on the header
                 actionHistory.trackActionSenderInfo(sender.fullName, payloadName)
                 val validatedRequest = validateRequest(request)
-                // removes incoming '#' if included in separation characters
-                val content = validatedRequest.content.replace("|^~\\&#", "|^~\\&")
-                val rawBody = content.toByteArray()
 
                 // if the override parameter is populated, use that, otherwise use the sender value
-                val allowDuplicates = if
-                (!allowDuplicatesParam.isNullOrEmpty()) allowDuplicatesParam == "true"
-                else {
-                    sender.allowDuplicates
-                }
+                val allowDuplicates =
+                    if (!allowDuplicatesParam.isNullOrEmpty())
+                        allowDuplicatesParam == "true"
+                    else
+                        sender.allowDuplicates
 
                 // Only process the report if we are not checking for connection or validation.
                 if (option != Options.CheckConnections && option != Options.ValidatePayload) {
                     val receiver = SubmissionReceiver.getSubmissionReceiver(sender, workflowEngine, actionHistory)
-
+                    val content =
+                        if (receiver is UniversalPipelineReceiver) validatedRequest.content
+                        // removes incoming '#' if included in separation characters
+                        else validatedRequest.content.replace("|^~\\&#", "|^~\\&")
```

Comment:
I'm not 100% understanding this change?  Is this being added out of an abundance of caution?

Reply:
I'm also curious about this, but maybe for a different reason. If we take my suggestion above and always copy the value of MSH to our FHIR extension, then we can just always right the MSH info from our FHIR extension to the output HL7v2 automatically and not need any extra logic like this?

Reply:
This line intends to preserve existing behavior in this path for the legacy pipeline. The code to handle the five-char encoding is only in the Universal Pipeline, so without this, any five-char encoding strings sent to the legacy pipeline will lead to failures.